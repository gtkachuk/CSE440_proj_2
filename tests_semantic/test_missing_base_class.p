program testMissingBaseClass;

class AA

BEGIN
   VAR row : integer;
END

class BB extends AAAA

BEGIN
   VAR column : integer;
END

class testMissingBaseClass extends BB

BEGIN
   
   VAR test: integer;

FUNCTION testMissingBaseClass;
BEGIN
   test := 8
END

END
.
