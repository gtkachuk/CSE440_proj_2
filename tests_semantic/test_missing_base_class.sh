#!/bin/sh

. ../tests_lib.inc

test_diff_sem test_missing_base_class.p
