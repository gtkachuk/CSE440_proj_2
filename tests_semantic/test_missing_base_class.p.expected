ERROR:9:Class "BB" attempted to extend class "AAAA" which does not exist
Errors detected. Exiting.
